File: aula_04/c#/Log.cs
using System;
using System.IO;

namespace projeto
{
    class Log
    {
        private string file { get; set; }

        public Log(string file)
        {
            this.file = file;
        }

        public void writeLine(string payload)
        {
            File.AppendAllText(this.file, payload + "\n");
        }

        public string getLog()
        {
            return File.ReadAllText(this.file);
        }
    }
}

File: aula_04/php/Log.php
<?php

class Log
{
    /** @var string $file */
    private $file;

    /**
     * Constructor
     *
     * @param string $file
     */
    public function __construct(string $file)
    {
        $this->file = $file;
    }

    /**
     * Write in log file
     *
     * @param string $payload
     * @return void
     */
    public function writeLine(string $payload): void
    {
        file_put_contents($this->file, $payload . "\n", FILE_APPEND);
    }

    /**
     * Get log data
     *
     * @return string
     */
    public function getLog(): string
    {
        return file_get_contents($this->file);
    }
}
